$Id: fix_warnings.patch 379 2012-09-20 11:45:26Z tg $

Fix code:
• array_merge expects at least one argument
• apparently, checking for syntax with unserialize() needs an extra

--- a/includes/SkinTemplate.php
+++ b/includes/SkinTemplate.php
@@ -1808,7 +1808,7 @@ abstract class BaseTemplate extends Quic
 			}
 		}
 
-		if ( $option == 'flat' ) {
+		if ( $option == 'flat' && $validFooterLinks ) {
 			// fold footerlinks into a single array using a bit of trickery
 			$validFooterLinks = call_user_func_array(
 				'array_merge',
--- a/includes/media/PNG.php
+++ b/includes/media/PNG.php
@@ -78,7 +78,7 @@ class PNGHandler extends BitmapHandler {
 		}
 
 		wfSuppressWarnings();
-		$data = unserialize( $metadata );
+		$data = @unserialize( $metadata );
 		wfRestoreWarnings();
 
 		if ( !$data || !is_array( $data ) ) {
